### ISSUE TITLE
The semantics of "set every" may be surprising to users

### ISSUE THREAD

**Issue by cunei** [sbt/sbt#1124]
The command "set every" redefines the tasks/settings that match the supplied key in all the scopes. Now, consider the following:

```
set every sources in (Compile,doc) := Seq.empty
```

The result is that the scope supplied in the command is silently ignored, and all the sources in every scope are emptied, leading to empty executable jars.

This semantics may be quite surprising where users assume that "set every" just iterates over all the projects; the fact that the supplied scope is silently discarded may be cause for considerable confusion.

Possible approaches: "set every" should either stop if a scope is supplied in the command line, or at least issue a warning. As a more complete solution, "set every" could be changed in order to take some scope filtering into account. If a change in semantics is not viable, an alternate command "set-in-all-projects" could be added in order to supply the alternate semantics that users may expect.

**Comment by eed3si9n** [comment]
To potential contributor: Likely the relevant code base is 
https://github.com/sbt/sbt/blob/v1.1.5/main/src/main/scala/sbt/internal/SettingCompletions.scala#L39-L65 where `setAll` is implemented.

https://github.com/sbt/sbt/blob/v1.1.5/main/src/main/scala/sbt/Main.scala#L408-L431 might also be involved.

**Comment by AkshatJindal1** [comment]
I would like to work on this issue

**Comment by eed3si9n** [comment]
@AkshatJindal1 Awesome!

**Comment by AkshatJindal1** [comment]
I have created a PR where I have added a log statement in case argument list is provided.
https://github.com/sbt/sbt/pull/6560

I am not able to add any reviewer to my PR. @eed3si9n can you have a look into it

**Comment by SethTisue** [comment]
the linked PR appears abandoned

**Comment by Crystora** [comment]
@eed3si9n Can I pick this up?

**Comment by eed3si9n** [comment]
@bittoby Yes please. Please check out our [Contributor's Guildeline](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) and:
1. Make sure you can reproduce the problem and the fix manually using locally built `sbt`
2. Include tests for the change
3. Follow our AI assisted contributions guideline (please disclose AI usages in the PR description, if any)
4. Multiple contributors might work on this issue in parallel
5. Generally we recommend getting CI to pass on a forked repo before sending a pull request (pull requests that do not pass tests will be closed after a few days)
6. Sign Scala CLA <https://contribute.akka.io/contribute/cla/scala>
